### PR TITLE
Support customizing VM disks and mounting remote VMs in `tart run`

### PR DESCRIPTION
**Pull request by edigaryev**
To create a VM with a custom disk:

```shell
tart create --linux xcode-16.0
tart set --disk <path to DMG> xcode-16.0
```

To mount this VM after pushing (and pulling it):

```shell
tart run --disk ghcr.io/cirruslabs/xcode:16.0 <VM>
```

### CODE REVIEW

File: Sources/tart/Commands/Run.swift [hunk 1]
```diff
@@ -495,6 +499,25 @@ struct Run: AsyncParsableCommand {
         continue
       }
 
+      // Support remote VM names in --disk command-line argument
+      if let remoteName = try? RemoteName(diskPath) {
+        let vmDir = try VMStorageOCI().open(remoteName)
+
+        // Unfortunately, VZDiskImageStorageDeviceAttachment does not support
+        // FileHandle, so we can't easily clone the disk, open it and unlink(2)
+        // to simplify the garbage collection, so use the <>
```
Comment (fkorotkov) [inline Sources/tart/Commands/Run.swift:508]: Can we mount the same disk to two VMs? Maybe only in read-only mode?

Comment (edigaryev) [reply Sources/tart/Commands/Run.swift:508]: Yes, but that will violate the OCI cache immutability guarantees that we currently have and still doesn't sound very secure compared to having a separate FS object.

File: Sources/tart/Commands/Run.swift [hunk 2]
```diff
@@ -495,6 +499,25 @@ struct Run: AsyncParsableCommand {
...
       }
 
+      // Support remote VM names in --disk command-line argument
+      if let remoteName = try? RemoteName(diskPath) {
+        let vmDir = try VMStorageOCI().open(remoteName)
+
+        // Unfortunately, VZDiskImageStorageDeviceAttachment does not support
+        // FileHandle, so we can't easily clone the disk, open it and unlink(2)
+        // to simplify the garbage collection, so use an intermediate directory.
+        let clonedDiskURL = try Config().tartTmpDir.appendingPathComponent("run-disk-\(UUID().uuidString)")
```
Comment (fkorotkov) [inline Sources/tart/Commands/Run.swift:509]: Should we skip it in `ro` mode?

Comment (edigaryev) [reply Sources/tart/Commands/Run.swift:509]: Answered in https://github.com/cirruslabs/tart/pull/847#discussion_r1653093093.

**Review by fkorotkov — APPROVED** [review]

**Pull request merged**